src/services/lnd/onchain-service.ts

```diff
@@ -25,19 +25,20 @@ export const OnChainService = (
     try {
       const { current_block_height } = await getHeight({ lnd })
       const after = Math.max(0, current_block_height - scanDepth) // this is necessary for tests, otherwise after may be negative
-      const { transactions } = await getChainTransactions({ lnd, after })
+      const { transactions }: GetChainTransactionsResult = await getChainTransactions({
+        lnd,
+        after,
+      })
 
       return transactions
-        .filter((tx) => !tx.is_outgoing || !!tx.transaction)
-        .map((tx) => {
+        .filter((tx) => !tx.is_outgoing && !!tx.transaction && !!tx.fee && tx.fee >= 0)
```

Comment:
in regards to documentation fee is number so `!!tx.fee` will return false if fee is zero, I know is an exceptional case but happened to me (I setup a receiving address in a mining pool that included its own transactions with zero fee).

Also I just check `null >= 0` in node console and is returning `true` --_-- so we cant use it. So maybe we should use default value before pass it to toSats or use something like `(tx.fee || 0) >= 0`

Reply:
Added a unit test to handle all cases